### PR TITLE
Closes 338. Map the DID string to the correct DID format

### PR DESCRIPTION
**Pull request by nick-verida**

### CODE REVIEW

File: packages/vda-did-resolver/src/getDIDs.ts [hunk 1]
```diff
@@ -28,13 +28,21 @@ export async function getDIDs(network: string, startIndex: number = 0, count: nu
...
     }
     
-    let data;
+    let data: string[] | undefined = undefined;
     try {
-        data = await contract.callStatic.getDIDs(startIndex, count)
+        data = (await contract.callStatic.getDIDs(startIndex, count)) as string[];
+        data.map((did) => {
+            // map the raw string into the correct canonoical DID format
+            return `did:vda:${network}:${did}`;
```
Comment (aurelticot) [inline packages/vda-did-resolver/src/getDIDs.ts:44]: Do we have a centralised util function for this formatting?

File: packages/vda-did-resolver/src/getDIDs.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { activeDIDCount } from "./activeDIDCount";
  * @param startIndex Index to start
  * @param count Total number of results to fetch
  */
-export async function getDIDs(network: string, startIndex: number = 0, count: number=20, mostRecent: boolean=true): Promise<string[]> {
+export async function getDIDs(network: string, startIndex: number=0, count: number=20, mostRecent: boolean=true): Promise<string[]> {
```
Comment (aurelticot) [inline packages/vda-did-resolver/src/getDIDs.ts:13]: Consider having a stronger type on `network`

File: packages/vda-did-resolver/src/getDIDs.ts [hunk 3]
```diff
@@ -28,13 +28,21 @@ export async function getDIDs(network: string, startIndex: number = 0, count: nu
...
     
-    let data;
+    let data: string[] | undefined = undefined;
     try {
-        data = await contract.callStatic.getDIDs(startIndex, count)
+        data = (await contract.callStatic.getDIDs(startIndex, count)) as string[];
+        data.map((did) => {
+            // map the raw string into the correct canonoical DID format
+            return `did:vda:${network}:${did}`;
+        })
```
Comment (aurelticot) [inline packages/vda-did-resolver/src/getDIDs.ts:45]: `map` doesn't mutate the array, so the `return data` is not modified as intended.

I suggest the following:

```ts
    // Move the try at the very top as the catch re-throws a generic error with a cause
    try {

        // ... 

        const data = (await contract.callStatic.getDIDs(startIndex, count)) as string[];
        // TODO: Consider validating `data` before processing and then removing the `as string[]` assertion
        // In the meantime, have a fall back with `data || []`, not the best though
        return (data || []).map((did) => `did:vda:${network}:${did}`)
    } catch (error: unknown) {
        throw new Error('Failed to get list of active DIDs', { cause: error });
    }
}
```

File: packages/vda-did-resolver/src/getDIDs.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { activeDIDCount } from "./activeDIDCount";
  * @param startIndex Index to start
  * @param count Total number of results to fetch
  */
-export async function getDIDs(network: string, startIndex: number = 0, count: number=20, mostRecent: boolean=true): Promise<string[]> {
+export async function getDIDs(network: string, startIndex: number=0, count: number=20, mostRecent: boolean=true): Promise<string[]> {
     const rpcUrl = RPC_URLS[network]
```
[the remark this answers is not in the document]
Comment (nick-verida) [reply packages/vda-did-resolver/src/getDIDs.ts:14]: agree

File: packages/vda-did-resolver/src/getDIDs.ts (hunk 1 above)
Comment (nick-verida) [reply packages/vda-did-resolver/src/getDIDs.ts:44]: Not that I've seen anywhere

File: packages/vda-did-resolver/src/getDIDs.ts (hunk 2 above)
Comment (nick-verida) [reply packages/vda-did-resolver/src/getDIDs.ts:13]: Agree. Fixing

File: packages/vda-did-resolver/src/getDIDs.ts (hunk 3 above)
Comment (nick-verida) [reply packages/vda-did-resolver/src/getDIDs.ts:45]: Agree

**Review by aurelticot — APPROVED** [review]
LGTM

**Pull request merged**